### PR TITLE
SRS - use get_pattern_subgroups_from_string

### PR DESCRIPTION
**Pull request by lbesnard**

### CODE REVIEW

File: aodndata/srs/srs_ghrsst.py [hunk 1]
```diff
@@ -2,19 +2,21 @@
...
 from datetime import datetime
 
+from aodncore.util.misc import get_pattern_subgroups_from_string
+
 from aodncore.pipeline import HandlerBase
 from aodncore.pipeline.exceptions import InvalidFileNameError
 
 L3S_L3C_FILE_PATTERN = re.compile(r"""
-                                ^(?P<nc_time_cov_start>[0-9]{14})-ABOM-
+                                (.*/|)(?P<nc_time_cov_start>[0-9]{14})-ABOM-
```
Comment (lwgordonimos) [inline aodndata/srs/srs_ghrsst.py]: What was the reason for this change to the pattern? Looks a bit odd to me...

Comment (lwgordonimos) [reply aodndata/srs/srs_ghrsst.py]: https://github.com/aodn/python-aodndata/pull/52/files#diff-62a887b1575ecffb84e3cff79877c85eR42

But you are already grabbing the basename here for this purpose?

Comment (lbesnard) [reply aodndata/srs/srs_ghrsst.py]: good point, but I should change it here 
https://github.com/aodn/python-aodndata/pull/52/files#diff-62a887b1575ecffb84e3cff79877c85eR45

**Comment by lbesnard** [comment]
@lwgordonimos rectified

File: aodndata/srs/srs_oc_gridded.py [hunk 2]
```diff
@@ -14,32 +15,21 @@
 OC_GRIDDED_PREFIX_PATH = 'IMOS/SRS/OC/gridded'
 
 RJOHNSON_FILE_PATTERN = re.compile(r"""
-                                ^(?P<data_parameter_code>A|S)
+                                (?P<data_parameter_code>A|S)
```
Comment (lwgordonimos) [inline aodndata/srs/srs_oc_gridded.py]: Not sure what the best practice is with the carets when using re.match... they seem redundant, but the pattern is separate from the match function, so I wonder if we should keep them in?

**Pull request merged**